cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -491,6 +530,12 @@ public ClusterModel clusterModel(long from,
           clusterModel.setBrokerState(brokerId, Broker.State.BAD_DISKS);
         }
       }
+
+      // Populate replica placement information for the cluster model.
+      if (populateReplicaPlacementInfo) {
```

Comment:
Can we pass `populateReplicaPlacementInfo` to `populateLoad` to be able to send original disk to `new Replica(...)` upon creation? This will ensure that the order we create cluster components and replicas is always consistent by construction.